### PR TITLE
Document and test that caller() does not see try{} blocks 

### PR DESCRIPTION
**Pull request by leonerd**
As per #18628

### CODE REVIEW

File: pod/perlfunc.pod [hunk 1]
```diff
@@ -989,6 +989,11 @@ L<C<undef>|/undef EXPR> if L<C<%^H>|perlvar/%^H> was empty.  Do not
 modify the values of this hash, as they are the actual values stored in
 the optree.
 
+Note that the only types of call frames that are visible are subroutine
+calls and C<eval>. Other forms of context, such as C<while> or C<foreach>
+loops or C<try> blocks are not considered interesting to C<caller>, as they
+do not alter the behaviour of the C<return> expression.
```
Comment (happy-barney) [inline pod/perlfunc.pod:995]: I'd use (your) sentence "caller() tells you where return would go".

It feels much more cleaner.

**Comment by leonerd** [comment]
@dur-randir You have downvoted the initial comment but not given any indication why.

Could you perhaps give some more wording here to explain your thoughts?

**Comment by dur-randir** [comment]
>Could you perhaps give some more wording here to explain your thoughts?


It's on the list discussion.

**Comment by leonerd** [comment]
@dur-randir 
> It's on the list discussion.

I see; so it is less that you are objecting to the tests and documentation this PR in itself, and more the underlying behaviour that they describe? That behaviour is in bleadperl already - if you think it should be changed perhaps you could make a counter-proposal and suggest that?

This particular PR is only documenting and testing behaviour which is *already* in the code - objecting to this PR won't prevent the behaviour from occurring.

**Comment by Leont** [comment]
If I understand things correctly, `caller` and `return` are in agreement. That sounds fine by me.

**Comment by leonerd** [comment]
@Leont 
> If I understand things correctly, `caller` and `return` are in agreement. That sounds fine by me.

Indeed they are. This PR just adds some documentation and unit-tests for that fact. I didn't think about `caller` in my original implementation, and observed it just happens to work by happy coïncidence.

**Comment by shadowcat-mst** [comment]
@dur-randir caller() seems to generally return 'the thing we would return from' - so eval {} shows up as a frame because 'return' inside an eval only returns from the eval. OTOH 'return' inside a try returns from the containing subroutine, so that's the frame. 

So if https://www.nntp.perl.org/group/perl.perl5.porters/2021/03/msg259257.html is your objection, then I think you're mistaken - they're *not* semantically identical and while they *do* have different behaviour, that's considered a feature of 'try' and both of them behave the same wrt return/caller - as does basically everything else, because caller/return share the relevant logic in the internals so to make them disagree with each other would require special casing one of the two.

As such, I think your worries are well-intentioned but misplaced.

**Pull request merged**